src/transformers/models/led/tokenization_led.py

```diff
@@ -40,18 +49,378 @@
 }
 
 
-class LEDTokenizer(BartTokenizer):
+@lru_cache()
+# Copied from transformers.models.bart.tokenization_bart.bytes_to_unicode
+def bytes_to_unicode():
+    """
+    Returns list of utf-8 byte and a mapping to unicode strings. We specifically avoids mapping to whitespace/control
+    characters the bpe code barfs on.
+
+    The reversible bpe codes work on unicode strings. This means you need a large # of unicode characters in your vocab
+    if you want to avoid UNKs. When you're at something like a 10B token dataset you end up needing around 5K for
+    decent coverage. This is a significant percentage of your normal, say, 32K bpe vocab. To avoid that, we want lookup
+    tables between utf-8 bytes and unicode strings.
     """
-    Construct a LED tokenizer.
+    bs = (
+        list(range(ord("!"), ord("~") + 1)) + list(range(ord("¡"), ord("¬") + 1)) + list(range(ord("®"), ord("ÿ") + 1))
+    )
+    cs = bs[:]
+    n = 0
+    for b in range(2**8):
+        if b not in bs:
+            bs.append(b)
+            cs.append(2**8 + n)
+            n += 1
+    cs = [chr(n) for n in cs]
+    return dict(zip(bs, cs))
 
-    [`LEDTokenizer`] is identical to [`BartTokenizer`] and runs end-to-end tokenization: punctuation splitting and
-    wordpiece.
 
-    Refer to superclass [`BartTokenizer`] for usage examples and documentation concerning parameters.
+# Copied from transformers.models.bart.tokenization_bart.get_pairs
+def get_pairs(word):
     """
+    Return set of symbol pairs in a word.
+
+    Word is represented as tuple of symbols (symbols being variable-length strings).
+    """
+    pairs = set()
+    prev_char = word[0]
+    for char in word[1:]:
+        pairs.add((prev_char, char))
+        prev_char = char
+    return pairs
+
+
+# Copied from transformers.models.bart.tokenization_bart.BartTokenizer with Bart->LED,BART-LED,facebook/bart-base->allenai/led-base-16384
```

Comment:
```suggestion
```
This one needs to be removed (or it overwrites all your other changes :-) )

Reply:
oh right we don't need this anymoreD: